packages/lib/src/spatial-navigation/components/virtualizedList/helpers/getNumberOfItemsVisibleOnScreen.ts

```diff
@@ -0,0 +1,24 @@
+const getMinSizeOfItems = <T>(data: T[], itemSize: number | ((item: T) => number)) => {
+  if (typeof itemSize === 'number') {
+    return itemSize;
+  }
+
+  let minSize = itemSize(data[0]); // Initialize with the size of the first item
```

Comment:
gestion d'erreur si data est vide ? C'est bizarre que ça te fasse pas d'erreur TS ici 🤔

Reply:
because we forgot to add `noUncheckedIndexedAccess` in the tsconfig 😭 
good point, thanks!

Reply:
I added a check to prevent this (even if the rule is not enforced for now)

Reply:
I think the function has become very convoluted 🙈 
I think you could've kept your old code but add an early return if the array is empty!

Reply:
We can talk about it this afternoon, I guess you did it this way because you temporarily enabled TS's check and made the changes accordingly. It's okay, but it's a shame that the code has more cognitive complexity now! Let's see if we can improve it 😁